### DISCUSSION TITLE
[Node Operator Question] Optimism Mainnet Node RPC issues: block delays and batched blocks

### DISCUSSION THREAD

**Discussion by wrinkledeth** [ethereum-optimism/developers#597] [Node Operators]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I followed this [guide](https://docs.optimism.io/builders/node-operators/tutorials/node-from-docker) and was able to sync an optimism mainnet node using op-node / op-geth.

when making rpc requests, very often, the node will "hang", and get stuck on the same block, and then all of a sudden batch together all of the pending blocks and jump ahead
here is a sample to explain what I mean 

![image](https://github.com/user-attachments/assets/613198c1-3151-4f75-bf13-ca18e230bed9)

I fetched  the latest block number every ~4 seconds in the above example
Do you know how I can solve /debug this issue? CPU, MEM, and disk consumption is low and normal

### Protocol Description

op-geth:v1.101408.0
op-node:v1.9.1
network name: OP Mainnet
Chainid: 10
L1 Chain: Eth Mainnet

**Comment by imtipi** [comment]
when do you screenshot this?
op mainnet produce blocks every 2 sec,you can get same block number in 2sec, and possible longer depend on network conditions(the sync)

**Reply by wrinkledeth** [reply]
This was screenshotted a few days ago. I get a block number every 2 seconds on other RPC's (alchemy). The node that I sync'd above usually produces a block every 2 seconds, but every ~30 seconds or so, I get the above lag, and the batched blocks type of odd behavior. Any ideas?

**Comment by Chomtana** [comment]
Do you notice such a delay on op-node too?

```
curl -X POST -H "Content-Type: application/json" http://127.0.0.1:8547 --data '{"jsonrpc":"2.0","method":"optimism_syncStatus","params":[],"id":1}' | jq
```

This command should query op-node status including block number that should be matched with op-geth

**Reply by wrinkledeth** [reply]
I ended up doing a resync and the problem went away. I will update this thread again with better logs if I ever notice the same issue.
